### DISCUSSION TITLE
All 'merge' commands failing

### DISCUSSION THREAD

**Discussion by StevenXL** [git-town/git-town#3403] [Q&A] [answered]
Hi,

When I run any command, such as `git sync` or `git append`, `git-town` will run `git merge --no-edit main` at some point. 

All of these are failing with:

```
fatal: Not possible to fast-forward, aborting.
```

It I add `--no-ff` to the command `git-town` is running, things work. Is appending this something I can ask `git-town` to do automatically? Is that a dangerous thing to do?

**Comment by kevgo** [comment]
Thanks for reaching out about this. Git Town v14.1 adds the `--ff` switch to the merge commands to fix issues that users who have [merge.ff](https://git-scm.com/docs/git-merge#Documentation/git-merge.txt---ff) set to false. This seems to conflict with your setup. To investigate this issue, can you please post the output of `git config -l`? Please anonymize confidential info like tokens. Thanks!

**Reply by StevenXL** [reply]
I can confirm that I am using `Git Town 14.1.0`. 

I am attaching the output of `git config -l`. Thank you for warning me about the token. I'm OK with my email / name out there.

[file.txt](https://github.com/git-town/git-town/files/15087393/file.txt)

**Reply by StevenXL** [reply]
Ah - it looks like I have `merge.ff=only`, and from the manual:

```
With --ff-only, resolve the merge as a fast-forward when possible. When not possible, refuse to merge and exit with a non-zero status.
```

So it can't do a fast-forward, so it fails.

**Reply by kevgo** [reply, marked as answer]
Excellent, thanks for helping investigate this issue! I have created https://github.com/git-town/git-town/pull/3408 that should fix this.

Please note that this fix relaxes the `ff-only` rules you have set for merging parent and tracking branches to use merge commits if needed. Branches can get so much out of sync that it's not always possible to fast-forward them. If this doesn't work for you, you could try setting the [sync-feature-strategy](https://www.git-town.com/preferences/sync-feature-strategy) to `rebase`. Git Town will then rebase your feature branches against their parent and tracking branches, thereby avoiding merge commits.

**Reply by kevgo** [reply]
Note: this is not because of the new `--ff` switch but because of the _lack_ of it in other Git commands run by Git Town. The concurrent timing with the rollout of v14.1 seems just coincidence.

**Reply by StevenXL** [reply]
Thank you - and let me say that I am very glad I found this tool. It's pretty awesome - particularly it's `stacked diffs` support; game-changer for my workflow!

**Reply by kevgo** [reply]
Thanks for saying that - it really means a lot! Please help spread the word 🙂

**Reply by kevgo** [reply]
Git Town [v14.2](https://github.com/git-town/git-town/releases/tag/v14.2.0) contains the fix. Please give it a try and reach out if you have any other feedback or suggestions.
